internal/tui/components/chat/messages/tool.go

```diff
@@ -34,15 +36,13 @@ type ToolCallCmp interface {
 	GetToolResult() message.ToolResult // Access to tool result data
 	SetToolResult(message.ToolResult)  // Update tool result
 	SetToolCall(message.ToolCall)      // Update tool call
-	SetCancelled()                     // Mark as cancelled
 	ParentMessageID() string           // Get parent message ID
 	Spinning() bool                    // Animation state for pending tools
 	GetNestedToolCalls() []ToolCallCmp // Get nested tool calls
 	SetNestedToolCalls([]ToolCallCmp)  // Set nested tool calls
 	SetIsNested(bool)                  // Set whether this tool call is nested
```

Comment:
Split brain!
GetNestedToolCalls() can be non-empty while SetIsNested() return `true`

Reply:
Okay, I got something wrong here.
GetNestedToolCalls gets nested ToolCalls of this instance, while SetIsNested() marks this instance as a Nested ToolCall itself.